winpr/libwinpr/clipboard/synthetic_file.c

```diff
@@ -269,34 +247,35 @@ static WCHAR* concat_remote_name(const WCHAR* dir, const WCHAR* file)
 		return NULL;
 
 	memcpy(buffer, dir, len_dir * sizeof(WCHAR));
-	buffer[len_dir] = L'\\';
+	buffer[len_dir] = L'/';
 	memcpy(buffer + len_dir + 1, file, len_file * sizeof(WCHAR));
 	return buffer;
 }
 
-static BOOL add_file_to_list(wClipboard* clipboard, const char* local_name,
+static BOOL add_file_to_list(wClipboard* clipboard, const WCHAR* local_name,
                              const WCHAR* remote_name, wArrayList* files);
 
-static BOOL add_directory_entry_to_list(wClipboard* clipboard, const char* local_dir_name,
-                                        const WCHAR* remote_dir_name, const struct dirent* entry,
-                                        wArrayList* files)
+static BOOL add_directory_entry_to_list(wClipboard* clipboard, const WCHAR* local_dir_name,
+                                        const WCHAR* remote_dir_name,
+                                        const LPWIN32_FIND_DATAW pFileData, wArrayList* files)
 {
 	BOOL result = FALSE;
-	char* local_name = NULL;
+	WCHAR* local_name = NULL;
 	WCHAR* remote_name = NULL;
 	WCHAR* remote_base_name = NULL;
 
 	/* Skip special directory entries. */
-	if ((strcmp(entry->d_name, ".") == 0) || (strcmp(entry->d_name, "..") == 0))
+
+	if ((_wcscmp(pFileData->cFileName, L".") == 0) || (_wcscmp(pFileData->cFileName, L"..") == 0))
```

Comment:
`Be careful with `L'somestring'`, windows understands `UCS-2/UTF16` most other systems `UCS4/UTF32`